### PR TITLE
Routes -> Controllers -> Model fo r User Entity and Mongo DB set up

### PR DESCRIPTION
**Pull request by AndyLiang1**
Created the Router -> Controller -> Model Pattern for User. This PR also contains code to connect to our mongoDB.

Router calls controllers and is responsible for only exposing the route.
Controller contains business logic and accesses the DB. Controllers may also call other controllers.
Models define the schema of a collection (think of a table in SQL).

To test this PR, you need a .env file in the server folder. See the secrets channel.
In postman or google, put in http://localhost:5000/api/user to see the result.

Also, the Atlas Cluster now has all IPs whitelisted.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "dependencies": {
+    "mongodb": "^5.5.0",
+    "mongoose": "^7.2.1"
+  }
+}
```
Comment (NLmeng) [inline package.json:6]: might have accidentally install the dependencies in root. should be fine though. will be resolved in my pr

File: server/config/config.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require('dotenv').config()
+
+const MONGO_USERNAME = process.env.MONGO_USERNAME || ''
+const MONGO_PASSWORD = process.env.MONGO_PASSWORD || ''
+const MONGO_URI = `mongodb+srv://${MONGO_USERNAME}:${MONGO_PASSWORD}@cluster0.4bflh41.mongodb.net/BugStormDB?retryWrites=true&w=majority`
+const SERVER_PORT = process.env.PORT ? Number(process.env.PORT) : 5001
```
Comment (NLmeng) [inline server/config/config.js:6]: i changed this to use port 5001, some mac has problems with port 5000 https://developer.apple.com/forums/thread/682332

**Review by NLmeng — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (AndyLiang1) [reply package.json:6]: Oh yea oops I only removed the node module

File: server/config/config.js (hunk 2 above)
Comment (AndyLiang1) [reply server/config/config.js:6]: No problem :)

**Comment by L0Lmaker** [comment]
We should also add local development option. Will create a new Issue for it.

**Review by L0Lmaker — APPROVED** [review]
LGTM

**Pull request merged**